packages/helpwave_widget/lib/src/shapes/circle.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:flutter/material.dart';
+import 'package:helpwave_theme/constants.dart';
+
+/// A [Widget] Shaped like a circle
+class Circle extends StatelessWidget {
+  /// The radius of the circle
+  final double diameter;
+
+  /// The [Color] of th circle
+  final Color? color;
+
+  const Circle({super.key, this.diameter = distanceSmall, this.color});
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: diameter,
+      height: diameter,
+      decoration: BoxDecoration(
+        shape: BoxShape.circle,
+        color: color ?? Colors.black,
```

Comment:
Why don't you define the default value in the constructor?

Reply:
Not assigning the value in the constructor allows better reusability by allowing the attribute to be optional. Otherwise Nullchecks and assignments would be needed, which disable the constructor from being const

Reply:
Parameters with default values are optional per definition. If you don't provide a value the default one will be used.
You don't really need nullchecks on dart nowadays... Just change the `Color?` to `Color` and you're good.
Using the `??` operator with `Color?` used in a single loc is unnecessary.

Reply:
I don't mean optional in this sense of an optional parameter.

If color is of type `Color`, then another class cannot just pass down an optional argument like this, C would be our class
A(double? x) => B(x: x) => C(x:x)
but would require something like
A(double? x) => B(x: x ?? 2) => C(x: x ?? 3)

`const A({this.x}){x ??= 2;}` isn't allowed, because const Constructors can't have a body.